lib/copy/copy.js

```diff
@@ -2,51 +2,213 @@
 
 const fs = require('graceful-fs')
 const path = require('path')
-const ncp = require('./ncp')
-const mkdir = require('../mkdirs')
+const mkdirs = require('../mkdirs').mkdirs
+const utimes = require('../util/utimes').utimesMillis
 
-function copy (src, dest, options, callback) {
-  if (typeof options === 'function' && !callback) {
-    callback = options
+const DEST_NOENT = -1
+const DEST_EXISTS = 1
+
+function copy (src, dest, options, cb) {
+  if (typeof options === 'function' && !cb) {
+    cb = options
     options = {}
   } else if (typeof options === 'function' || options instanceof RegExp) {
     options = {filter: options}
   }
-  callback = callback || function () {}
+
+  cb = cb || function () {}
   options = options || {}
 
-  // Warn about using preserveTimestamps on 32-bit node:
+  // default to true for now
+  options.clobber = 'clobber' in options ? !!options.clobber : true
+  // overwrite falls back to clobber
+  options.overwrite = 'overwrite' in options ? !!options.overwrite : options.clobber
+  options.dereference = 'dereference' in options ? !!options.dereference : false
+  options.preserveTimestamps = 'preserveTimestamps' in options ? !!options.preserveTimestamps : false
+
+  options.filter = options.filter || function () { return true }
+
+  // Warn about using preserveTimestamps on 32-bit node
   if (options.preserveTimestamps && process.arch === 'ia32') {
     console.warn(`fs-extra: Using the preserveTimestamps option in 32-bit node is not recommended;\n
     see https://github.com/jprichardson/node-fs-extra/issues/269`)
```

Comment:
@RyanZim would you please say what you mean exactly? is it referring to the extra `\n` in the warn?